### PR TITLE
Resolve 300

### PR DESCRIPTION
**Pull request by datadavev**
Implements the resolver functionality for EZID.

### CODE REVIEW

File: ezidapp/management/commands/diag-identifier.py [hunk 1]
```diff
@@ -141,8 +159,70 @@ def add_arguments(self, parser:argparse.ArgumentParser):
...
+        )
+        _metrics.add_argument(
+            '-e',
+            '--end',
+            type=datetime.datetime.fromisoformat,
+            default=datetime.datetime.utcnow(),
+            help="Ending date for metrics"
+        )
+
+        _syncmeta = subparsers.add_parser(
```
Comment (jsjiang) [inline ezidapp/management/commands/diag-identifier.py:202]: What is the use case for this "syncmeta" option?

File: doc/resolver_dev.md [hunk 2]
```diff
@@ -1,20 +1,172 @@
...
+
+If the status is "R" ("Reserved"), a status code of 403 Forbidden is returned.
+
+If the status is "U" ("Unavailable"), a status code of 308 Permanent Redirect is returned with a Location header containing the tombstone page for that identifier. The tombstone page will typically be:
+
+```
+{BASE_URL}/tombstone/id/{identifier}
+```
+
+If the status is "P" ("Public"), a status code of 307 Temporary Redirect is returned with a Location header containing the registered Location value.
```
Comment (jsjiang) [inline doc/resolver_dev.md:161]: Is it 302 in this case?

File: doc/resolver_dev.md [hunk 3]
```diff
@@ -1,20 +1,172 @@
...
+
+If the value is not an EZID shoulder, the client is redirected to:
+
+```
+https://n2t.net/{identifier}
+```
+
+If the value is a shoulder managed by EZID:
+
+If the status is "R" ("Reserved"), a status code of 403 Forbidden is returned.
```
Comment (jsjiang) [inline doc/resolver_dev.md:153]: Is this 404?

File: doc/resolver_dev.md [hunk 4]
```diff
@@ -1,20 +1,172 @@
...
+
+If the status is "U" ("Unavailable"), a status code of 308 Permanent Redirect is returned with a Location header containing the tombstone page for that identifier. The tombstone page will typically be:
```
Comment (jsjiang) [inline doc/resolver_dev.md:155]: Is this 302?

Comment (datadavev) [reply doc/resolver_dev.md:155]: resolver_dev.md contains notes generated during the resolver service development. The document does not reflect the final state of implementation, instead please refer to the API documentation and source code for actual implementation details.

File: settings/settings.py.j2 [hunk 5]
```diff
@@ -209,6 +209,9 @@ LOGGING_CONFIG = None
 # Disable EZID's logging config (would be performed in impl.log)
 LOGGING_CONFIG_FILE = None
 
+# Default log_level to DEBUG if not specified.
+{% if log_level is not defined %}{% set log_level = 'DEBUG' %}{% endif %}
```
Comment (jsjiang) [inline settings/settings.py.j2:213]: We have a "log_level" parameter set AWS SSM as /uc3/ezid/ui/default/log_level. We use {{log_level}} to get the value in thesettings.py.j2. Are these two variables refer to the same thing?

File: impl/resolver.py [hunk 6]
```diff
@@ -0,0 +1,258 @@
...
+class ArkIdentifierValueParser(IdentifierValueParser):
+    # two chars after percent character
+    # https://www.ietf.org/id/draft-kunze-ark-36.html#name-normalization-and-lexical-e
+    # step 5
+    PERCENT_MATCH = re.compile("(%[a-zA-Z0-9]{2})")
+    STRUCTURAL_MATCH_1 = re.compile("//+")
+    STRUCTURAL_MATCH_2 = re.compile("[.][.]+")
+    STRUCTURAL_MATCH_3 = re.compile("[.]/")
+    STRUCTURAL_MATCH_4 = re.compile("/[.]")
+    INFLECTION_MATCH = re.compile("\?info|\?{2}|\?$")
```
Comment (jsjiang) [inline impl/resolver.py:183]: What are the expected behaviors for requests ends with a single "?"

File: tests/resolve/test_resolve.py [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+@pytest.mark.parametrize("val,expected",[
+    ('ark:/99166/',(404,"","")),
+    ('ark:/99166/p3wp9v20',(404,"","")),
+    ('ark:/99166/p3wp9v205',(302,"https://ezid.cdlib.org/id/ark:/99166/p3wp9v205","")),
+    ('ark:/99166/p3wp9v205?', (302, "https://ezid.cdlib.org/id/ark:/99166/p3wp9v205", "")),
+    ('ark:/99166/p3wp9v205??', (200, "", "")),
+    ('ark:/99166/p3wp9v205?info', (200, "", "")),
+    ('ark:/99166/p3wp9v205_extra', (302, "https://ezid.cdlib.org/id/ark:/99166/p3wp9v205_extra", "")),
+    ('ark:/99166/p3wp9v205%20extra', (302, "https://ezid.cdlib.org/id/ark:/99166/p3wp9v205%20extra", "")),
+    ('ark:/99166/p3wp9v20??', (200, "", "")),
```
Comment (jsjiang) [inline tests/resolve/test_resolve.py:39]: What to expect when request ends with a single '?'  'ark:/99166/p3wp9v20?'

**Pull request merged**